web3/providers/eth_tester/middleware.py

```diff
@@ -251,23 +253,51 @@ def is_named_block(value):
 })
 
 
+def guess_from(web3, transaction):
+    if web3.eth.coinbase:
+        return web3.eth.coinbase
+    elif web3.eth.accounts:
+        return web3.eth.accounts[0]
+    else:
+        return None
+
+
+def guess_gas(web3, transaction):
+    return web3.eth.estimateGas(transaction) * 2
+
+
+@curry
+def fill_default(field, guess_func, web3, transaction):
+    if field in transaction and transaction[field] is not None:
+        return transaction
+    else:
+        guess_val = guess_func(web3, transaction)
+        return assoc(transaction, field, guess_val)
+
+
+fill_default_from = fill_default('from', guess_from)
+fill_default_gas = fill_default('gas', guess_gas)
+
+
 def default_transaction_fields_middleware(make_request, web3):
     def middleware(method, params):
-        if method in (
-            'eth_call',
+        # TODO send call to eth-tester without gas, and remove guess_gas entirely
+        if method == 'eth_call':
+            filled_transaction = pipe(
+                params[0],
+                fill_default_from(web3),
```

Comment:
instead of currying these for each call, you could curry them in the outer scope of this function once during middleware construction.